File: 51-javascript-alerts/csharp/JavaScriptAlerts.cs
using NUnit.Framework;
using OpenQA.Selenium;
using OpenQA.Selenium.Firefox;

public class JavaScriptAlerts
{
    IWebDriver Driver;

    [SetUp]
    public void SetUp()
    {
        Driver = new FirefoxDriver();
    }

    [TearDown]
    public void TearDown()
    {
        Driver.Quit();
    }

    [Test]
    public void JavaScriptAlertAppearsAndAccepted()
    {
        Driver.Navigate().GoToUrl("http://the-internet.herokuapp.com/javascript_alerts");
        Driver.FindElement(By.CssSelector(".example li:nth-child(2) button")).Click();
        IAlert Popup = Driver.SwitchTo().Alert();
        Popup.Accept();
        string Result = Driver.FindElement(By.Id("result")).Text;
        Assert.That(Result.Equals("You clicked: Ok"));
    }

}

File: 51-javascript-alerts/python/javascript_alerts.py
# -*- coding: utf-8 -*-
"""
Implementation of http://elementalselenium.com/tips/51-javascript-alerts
"""

import unittest
from selenium import webdriver


class JavaScriptAlerts(unittest.TestCase):

    def setUp(self):
        self.driver = webdriver.Firefox()

    def tearDown(self):
        self.driver.quit()

    def test_example_1(self):
        driver = self.driver
        driver.get('http://the-internet.herokuapp.com/javascript_alerts')
        driver.find_elements_by_tag_name('button')[1].click()
        popup = driver.switch_to.alert
        popup.accept()
        result = driver.find_element_by_id('result').text
        assert result == 'You clicked: Ok'

if __name__ == "__main__":
    unittest.main()
